src/components/pages/History/PublishedList.tsx

```diff
@@ -4,33 +4,60 @@ import { useOcean } from '@oceanprotocol/react'
 import React, { ReactElement, useEffect, useState } from 'react'
 import Loader from '../../atoms/Loader'
 import AssetList from '../../organisms/AssetList'
+import axios from 'axios'
+import { queryMetadata } from '../../../utils/aquarius'
 
 export default function PublishedList(): ReactElement {
-  const { ocean, status, accountId } = useOcean()
+  const { accountId } = useOcean()
   const [queryResult, setQueryResult] = useState<QueryResult>()
   const [isLoading, setIsLoading] = useState(false)
+  const { config } = useOcean()
+  const source = axios.CancelToken.source()
+  const [page, setPage] = useState<number>(1)
 
   useEffect(() => {
     async function getPublished() {
-      if (!accountId || !ocean) return
-
+      if (!accountId) return
+      const queryPublishedAssets = {
+        page: page,
+        offset: 9,
+        query: {
+          nativeSearch: 1,
+          query_string: {
+            query: `(publicKey.owner:${accountId})`
+          }
+        },
+        sort: { 'price.ocean': -1 }
+      }
       try {
-        setIsLoading(true)
-        const queryResult = await ocean.assets.ownerAssets(accountId)
-        setQueryResult(queryResult)
+        queryResult || setIsLoading(true)
+        const result = await queryMetadata(
+          queryPublishedAssets as any,
+          config.metadataCacheUri,
+          source.token
+        )
+        setQueryResult(result)
       } catch (error) {
         Logger.error(error.message)
       } finally {
         setIsLoading(false)
       }
     }
     getPublished()
-  }, [ocean, status, accountId])
+  }, [accountId, page])
```

Comment:
this effect should fire when `config.metadataCacheUri` changes too, which usually happens when users switch networks in their wallet. So `config.metadataCacheUri` should be part of this dependency array

Reply:
right, I'll add that

Reply:
perfect, switching networks works now for showing different assets. Somewhere we also need to trigger an update to the pagination calculation, that didn't update for me on network change